zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java

```diff
@@ -118,6 +120,9 @@ public ZeppelinServer() throws Exception {
 
     notebook.addNotebookEventListener(heliumApplicationFactory);
     notebook.addNotebookEventListener(notebookWsServer.getNotebookInformationListener());
+
+    remoteWorksManager = new RemoteWorksManager(notebook);
+    replFactory.setRemoteController(remoteWorksManager.getInstance());
```

Comment:
`remoteWorksManager.getInstance()` is invoked here only, not anywhere else. 
That means we don't need singleton pattern here.